src/submodule.c

```diff
@@ -1941,7 +1943,17 @@ static int submodule_read_config(git_submodule *sm, git_config *cfg)
 		goto cleanup;
 	}
 
-	if ((error = get_value(&value, cfg, &key, sm->name, "ignore")) == 0) {
+	/* For `ignore', first check the config of the parent repository. */
+	if (git_repository_config_snapshot(&repo_snap_cfg, sm->repo) == 0) {
+		repo_cfg_cleanup = true;
+		if (get_value(&value, repo_snap_cfg, &key, sm->name, "ignore") == 0) {
+			override_ignore = true;
+		}
+	}
+
+	ign_cfg = override_ignore ? repo_snap_cfg : cfg;
+
+	if ((error = get_value(&value, ign_cfg, &key, sm->name, "ignore")) == 0) {
```

Comment:
It seems like we're looking up the value from `repo_snap_cfg`, then deciding whether we should look up from `repo_snap_cfg` or `cfg` - and then looking it up again?

I think the more natural pattern here would be looking up the value from `cfg` and then trying to look it up from the parent.  If that succeeds, we'll use it.  If not, continue using the one from the `cfg`.

Reply:
I have made a small change with the latest update while preserving the
lookup order. This way, we can use the value that is set in `gitmodules`
(like you suggested) and also return the right errors (if any) when
fetching values either from the parent or the submodule.

If you would still like the `gitmodules` lookup to happen first, I will
make that change in the next patch.

Thanks.